### ISSUE TITLE
Can not use room names that contains different charachaters for example åäö

### ISSUE THREAD

**Issue by small1** [MarcelCoding/jitsi-openid#266]
When you try to create a room containing dofferent char sets like åäö it failes with error 400

Try to create a room https://<yourinstance>/Räksmörgås  

Jitsi will go with it but the auth component will fail.

**Comment by MarcelCoding** [comment]
You did not got any additional context from jitsi openid? Logs or in the Browser? 

If i take the name exactly as you, jitsi is complaining:

![image](https://github.com/MarcelCoding/jitsi-openid/assets/34819524/079fe729-a064-46a0-b11d-345ee7c37990)

Maybe without the `:`? 

I've checked all the components in jitsi-openid individual that are doing something with the room name, and nothing should be failing with valid utf8 characters.

**Comment by small1** [comment]
Bahh. I did not proof read. It remove a part.  Try without https://

**Comment by small1** [comment]
So back to this again.  I have tested to start a jitsi meeting with swedish characters (or other ones)  and nginx says in the access log error 400.   openid does not say anything in the logs.

[30/Apr/2024:12:39:15 +0200] "GET /room/r%C3%A4ksm%C3%B6rg%C3%A5s HTTP/1.1" 400 0 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"

**Comment by MarcelCoding** [comment]
Sorry for the late response. I am currently waiting for a friend who is currently setting up a jitsi. I'll use that instance for debugging.

**Comment by small1** [comment]
@MarcelCoding if i give you access to environment with stuff would that help ? :)  I have quite a few servers up

**Comment by albig** [comment]
I decided to workaround this issue by replacing / removing all non-ASCII character already in nginx:

```
   location @root_path {
-    rewrite ^/(.*)$ / break;
+    content_by_lua_block {
+        local uri = ngx.unescape_uri(ngx.var.request_uri)
+
+      -- Replace Umlauts in the URL
+      uri = uri:gsub("ä", "ae")
+                :gsub("ö", "oe")
+                :gsub("ü", "ue")
+                :gsub("Ä", "Ae")
+                :gsub("Ö", "Oe")
+                :gsub("Ü", "Ue")
+                :gsub("ß", "ss")
+                :gsub("[^\32-\126]", "") -- remove all other, non-ASCII characters
+
+      -- Only redirect if the URL was changed
+      if uri ~= ngx.unescape_uri(ngx.var.request_uri) then
+          ngx.redirect(uri, 301)
+        else
+          ngx.exec('/')
+      end
+    }
```

This requires lua support in nginx. Which is provided by the `nginx-extras` package in Debian.

Maybe a workaround for others, too.

**Comment by small1** [comment]
The problem with that approach is that users that are not logged in will get to a completely different room as they are using a different url then the one that is authenticated.

**Comment by small1** [comment]
so in  src/router/index.js  we have base: generateUrl('/apps/contacts','')   This sets the baseurl for everything within contacts. Also the contact item  in the team.

**Comment by albig** [comment]
> The problem with that approach is that users that are not logged in will get to a completely different room as they are using a different url then the one that is authenticated.

No. All users are redirected. For sure a question, where you place it in the nginx config flow.
